rust/src/aggregator/rpc.rs

```diff
@@ -28,7 +28,43 @@ mod inner {
     }
 }
 
-pub use inner::{Rpc, RpcClient as Client};
+pub use inner::Rpc;
+
+#[cfg(not(test))]
+pub use inner::RpcClient as Client;
+
+#[cfg(test)]
+mod mocks {
+    pub use super::*;
+
+    use futures::future;
+    use mockall::mock;
+    use std::io;
+    use tarpc::{client::Config, context::Context, rpc::Transport};
+
+    mock! {
+        pub NewClient {
+            fn spawn(self) -> io::Result<MockClient>;
+        }
+    }
+
+    mock! {
+        pub Client {
+            fn new<T: Transport<(), ()> + 'static>(config: Config, transport: T) -> MockNewClient;
+            fn select(&mut self, ctx: Context, credentials: Credentials) -> future::Ready<io::Result<Result<(), ()>>>;
+            fn aggregate(&mut self, ctx: Context) -> future::Ready<io::Result<Result<(), ()>>>;
+        }
+    }
+
+    impl Clone for MockClient {
+        fn clone(&self) -> Self {
+            Self::default()
+        }
+    }
+}
+
+#[cfg(test)]
+pub use mocks::MockClient as Client;
```

Comment:
for readability, can we place the two cfg-dependent inputs of the `RpcClient` directly one below the other? this way it's clearer that they stand for a xor import of the same object under different configurations.

Reply:
Also a good point. I'll do that.